wayland-logout

```diff
@@ -0,0 +1,27 @@
+#!/bin/sh
+
+# REQUIREMENTS - lsof (not always present)
+
+if [ -z "${WAYLAND_DISPLAY}" ]; then
+	printf '%s\n' 'WAYLAND_DISPLAY not set'
+	exit 1
+fi
+
+case "${WAYLAND_DISPLAY}" in
+	/*)
+		SOCKET_PATH="${WAYLAND_DISPLAY}"
+		;;
+	*)
+		if [ -z "${XDG_RUNTIME_DIR}" ]; then
+			printf '%s\n' \
+				'WAYLAND_DISPLAY is not an absolute path and XDG_RUNTIME_DIR not set'
+			exit 2
+		fi
+		SOCKET_PATH="${XDG_RUNTIME_DIR}/${WAYLAND_DISPLAY}"
+		;;
+esac
+
+WAYLAND_PIDS="$(lsof -t -f -- ${SOCKET_PATH})"
+for WPID in $WAYLAND_PIDS; do
```

Comment:
Why does this handle potentially more than one pid? There should be only one pid per socket file.

Reply:
not necessarily, multiple processes can register themselves at a socket - https://stackoverflow.com/questions/670891/is-there-a-way-for-multiple-processes-to-share-a-listening-socket

Reply:
Well if this has happened somehow and there are multiple PIDs, I'd like to abort instead of killing all of them.

Reply:
done